src/dune_engine/scheduler.ml

```diff
@@ -782,17 +782,44 @@ end = struct
   ;;
 end
 
+module Trigger : sig
+  (** Conceptually, a [unit Fiber.Ivar.t] that may be filled more than once. *)
```

Comment:
More precisely, fills past the first one are ignored. So I would call this idempotent instead.

Reply:
Yes, good point. Note that since the data is unit, ignoring a fill is indistinguishable from not ignoring it. I've expanded the comment with more nuance. Feel free to edit it a bit if you prefer a different wording.